scripts/e2e/e2e.sh

```diff
@@ -94,6 +93,18 @@ export_base64_value() {
    export "$1"="$base64_value"
 }
 
+exit_cleanup_on_err() {
+  "$@"
```

Comment:
Hi @codenrhoden,

We really need to employ shellcheck on this script...Anyway, this should be rewritten as (please note this depends on the comment [above]()):

```shell
if ! eval "${@}"; then
  local exit_code="${?}"
  if [ "${VM_CREATED}" ]; then
    get_bootstrap_vm "${context}"
    delete_vm "${target_vm_prefix}"
  fi
  exit "${exit_code}"
fi
```